Added documentation standards (#19)
bcgov/citz-imb: standards/documentation.md

``` Boo
── Root-level README // Landing page, provides useful links.
── Non-root-level README // Technical directory details.
── Wiki // Information hub for everyone, especially non-developers.
```

<br />

</details>

## Table of Contents

- [Starting a Project](#starting-a-project) - **Start Here!**
- [What To Use READMEs For](#what-to-use-readmes-for) - Technical information.
- [What To Use The Wiki For](#what-to-use-the-wiki-for) - End-user / product oriented language.

<br />

***

## Starting a Project
Follow these guidelines when first starting a new project.

Create the basic root level README and basic Wiki pages before coding begins.
 - The basic README should include a link to your Wiki's home page, as well as a list of links to developer standards, contributing documents, and guidelines.

>  - The basic Wiki should include pages for user stories, personas, roadmaps, stakeholders, architecture diagrams, and entity relationship diagrams.

Reviewer: ```suggestion
 - The basic Wiki should include pages for user stories, personas, roadmaps, interested parties, architecture diagrams, and entity relationship diagrams.
```

├── src
|   ├── frontend // Decision logs, commands, explanation of directory structure. 
|   |   └── src
|   |       ├── components // Explanation of standards when writing and working with components, links to standards documents.
|   |       └── pages
|   └── backend // Decision logs, commands, explanation of directory structure. 
└── README.md // Landing page for repo, links to Wiki, links to API docs and environments,
              // standards documents and guidelines, explanation of directory structure. 
```

> [!NOTE]
> READMEs should link to standards documents.  
> It is suggested to keep standards/guidelines documents in either a standards directory at root level or in Wiki pages.

<br />

***

## What To Use The Wiki For
The Wiki should primarily be used to support everyone who isn't a developer. When a user wants to learn more about your application, you should direct them to the Wiki.

_Below is a list of required and potential Wiki pages:_

``` Boo
── Homepage (required) // Table of contents for the rest of the Wiki.
── Onboarding (required) // Steps to onboard a new developer. Includes starting tasks that new developers can try.

> ── Architecture (required) // From high-level to low-level diagrams of how the application works.

Reviewer: ```suggestion
── Architecture (required) // From high-level to low-level diagrams of how the application works.
    ── ERD // Entity Relationship Diagram and explanations of how each table is used.
    ── DAL // Data Access Layer or how ORM (Object Relational Mapper) works.
    ── ADR / ADL // Architectural Decision Records / Log
```

── Personas
── User Stories
── Roadmap

> ── Stakeholders

Reviewer: Also noted above, using 'interested parties' instead of 'stakeholders' is more inclusive language.

> ── Stakeholders

Author: ```suggestion
── Interested Parties
```

── ERD // Entity Relationship Diagram and explanations of how each table is used.

> ── DAL // Data Access Layer or how ORM (Object Relational Mapper) works.

Reviewer: These both probably belong within 'Architecture'

Author: Don't you think that'll become too large of a page?

Reviewer: I don't think that page size is a problem due to the power of the wiki's outline. I'm also think we should note the need for an ADL / ADR

Author: Would that not be part of the Architecture page?

Author: Could you put your idea into a suggestion?

> ── ERD // Entity Relationship Diagram and explanations of how each table is used.
> ── DAL // Data Access Layer or how ORM (Object Relational Mapper) works.

Reviewer: ```suggestion
```

> ── How To // Tutorials for the end-user on how to use the application.

Reviewer: This could potentially be encapsulated within 'Onboarding.' We could delineate between 'Developer' and 'User' within that parent context. Happy to discuss.

Author: ~~I'm worried about how large of a Wiki page onboarding would become. I think the onboarding page should definitely link to other pages such as DAL and and ERD. Same goes for the size of Onboarding if we include both user and developer.~~

Reviewer: Similar to the above regarding the 'Architecture' section, using the wiki gives us the power of the _outline_ and so I'm of the thought that page size isn't an issue. Correctly scoping information is of far higher priority, IMO.

Author: I can agree on the Architecture page but I still think Onboarding for Developers and End-Users should be separate as they are quite different.
